### ISSUE TITLE
unable to create feature report 

### ISSUE THREAD

**Issue by electroniceagle** [aloetesting/aloe_webdriver#23]
Pretty sure that I just don't know how to set this up.  I'm code diving right now, but what is the right way to get "And I capture a screenshot" configured?  I'm calling `set_save_directory(settings.PROJECT_PATH, 'aloe_html')` in my with_browser() method and the screenshots are getting generated.

``````
======================================================================
ERROR: test suite for <class 'aloe.testclass.Admin'>
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/bschott/.virtualenvs/sites/lib/python2.7/site-packages/nose/suite.py", line 228, in run
    self.tearDown()
  File "/Users/bschott/.virtualenvs/sites/lib/python2.7/site-packages/nose/suite.py", line 351, in tearDown
    self.teardownContext(ancestor)
  File "/Users/bschott/.virtualenvs/sites/lib/python2.7/site-packages/nose/suite.py", line 367, in teardownContext
    try_run(context, names)
  File "/Users/bschott/.virtualenvs/sites/lib/python2.7/site-packages/nose/util.py", line 471, in try_run
    return func()
  File "/Users/bschott/.virtualenvs/sites/lib/python2.7/site-packages/django/test/testcases.py", line 1288, in tearDownClass
    super(LiveServerTestCase, cls).tearDownClass()
  File "/Users/bschott/.virtualenvs/sites/lib/python2.7/site-packages/django/test/testcases.py", line 168, in tearDownClass
    super(SimpleTestCase, cls).tearDownClass()
  File "/Users/bschott/.virtualenvs/sites/lib/python2.7/site-packages/aloe/testclass.py", line 125, in tearDownClass
    cls.after_feature(cls.feature)
  File "/Users/bschott/.virtualenvs/sites/lib/python2.7/site-packages/aloe/registry.py", line 197, in after_func
    after_hook(*args, **kwargs)
  File "/Users/bschott/.virtualenvs/sites/lib/python2.7/site-packages/aloe_webdriver/screenshot.py", line 68, in record_run_feature_report
    os.path.basename(feature.described_at.file)
AttributeError: 'TestFeature' object has no attribute 'described_at'```
``````

**Comment by electroniceagle** [comment]
OK, maybe a bug.  I think description_node is the right attribute not 'described_at'.

```
(Pdb) p feature.description_node.__dict__
{'line': 2, 'lines': (u'This feature tests that the admin view functions correctly.',), 'col': 1, 'filename': '/Users/bschott/Source/sites/awesim/site_common/features/admin.feature'}
```

**Comment by koterpillar** [comment]
This is a bug, the screenshot feature got in without tests. I am curious to know your use case, since I'm using a much simpler "screenshot on failure" step instead.

**Comment by electroniceagle** [comment]
I hand hacked it just to use feature.filename and it generates what looks like the correct json file.  I'm not sure if there are cases where description_node.filename and feature.filename are different.

I'm was porting some code from our old lettuce terrain.py file that generates an html report and colorizes red or green with a link to the screenshot.  Sort of gives you the same experience as coverage report.

![screen shot 2016-04-07 at 7 26 02 pm](https://cloud.githubusercontent.com/assets/219202/14369890/cdc1cfac-fcf6-11e5-903d-61daa9081270.png)

We also had a single page that tiled all of the screenshot thumbnails so you could look for bad template bugs that didn't actually break lettuce (like at one point remember white text on white background suddenly became the default).  Lettuce was perfectly happy!

**Comment by electroniceagle** [comment]
I can PR that fix, but not sure about the right approach for the tests.  It appears we're really missing some command line flags to set that directory and I don't know why "./aloe_html" isn't just the default if you have --aloe_html=True or whatever.

**Comment by koterpillar** [comment]
Given that this feature is not very configurable in its current form, and broken for quite a long time, I'm going to remove it altogether. If you want to keep using the report, please copy it somewhere in your code base.

What I still want to add is a hook to save the page screenshot and HTML source on a failure.

**Comment by electroniceagle** [comment]
I'd like to see your screenshot on failure approach.  Seems faster and simpler, but isn't it almost the same amount of code?  I back burnered this particular feature until we get the rest of our lettuce integration tests ported.  I'd be happy to test it as a snippet if you care to share.  

We're trying to use this opportunity to clean up a lot of imperative non-Gherkin style integration tests and use the built-in steps wherever possible.  I might have several other minor issues to post where I got tripped up just for the sake of documentation for others.  For example,  "I see a link " steps doesn't find drop-down menus because of filter_displayed=True.  I made versions "I have a link " without this filter.  I also made versions that reverse on url name because we have multiple sites with different paths for the same thing (bike-shedding logout versus signout mostly :-).

**Comment by koterpillar** [comment]
Watch the pull requests here, I'll extract the failed screenshots once I have time.

If you have documentation problems, I'd like to hear about them - the docs are not as good as they can be, unfortunately.

In general, the steps here follow the user's perspective as much as possible. If a user does not _see_ a link, it will not be reported visible. If you want to check the menu for links, write your own step, such as "The menu contains a link to...".

**Comment by koterpillar** [comment]
The replacement screenshot steps are implemented in #25, that's about to be released.
